### DISCUSSION TITLE
Visual Studio Code on chromebook - dependency issues

### DISCUSSION THREAD

**Discussion by shgrrts** [microsoft/vscode-discussions#750] [Extension Development QnA] [answered]
I follow the steps to install VS code on Chromebook amd64 (https://code.visualstudio.com/blogs/2020/12/03/chromebook-get-started) and get stuck on double-click on the package to launch the installer.

I see: Install app with Linux Installation successfully started. Details Application: code Version: 1.81.0-1690979901 Description: Code editing. Redefined. Visual Studio Code is a new choice of tool that combines the simplicity of a code editor with what developers need for the core edit-build-debug cycle. See https://code.visualstudio.com/docs/setup/linux for installation instructions and FAQ.

I click OK and see Error installing Linux application

The following packages have unmet dependencies:

code: Depends: libasound2 (>= 1.0.17) but it is not installable

    Depends: libatk-bridge2.0-0 (>= 2.5.3) but it is not installable
    Depends: libatk1.0-0 (>= 2.2.0) but it is not installable
    Depends: libatspi2.0-0 (>= 2.9.90) but it is not installable
    Depends: libc6 (>= 2.17) but it is not installable
    Depends: libcairo2 (>= 1.6.0) but it is not installable
    Depends: libcurl3-gnutls but it is not installable or
             libcurl3-nss but it is not installable or
             libcurl4 but it is not installable or
             libcurl3 but it is not installable
    Depends: libdbus-1-3 (>= 1.0.2) but it is not installable
    Depends: libdrm2 (>= 2.4.60) but it is not installable
    Depends: libexpat1 (>= 2.0.1) but it is not installable
    Depends: libgbm1 (>= 17.1.0~rc2) but it is not installable
    Depends: libglib2.0-0 (>= 2.12.0) but it is not installable
    Depends: libglib2.0-0 (>= 2.39.4) but it is not installable
    Depends: libgssapi-krb5-2 but it is not installable
    Depends: libgtk-3-0 (>= 3.9.10) but it is not installable
    Depends: libgtk-3-0 (>= 3.9.10) but it is not installable or
             libgtk-4-1 but it is not installable
    Depends: libkrb5-3 but it is not installable
    Depends: libnspr4 (>= 2:4.9-2~) but it is not installable
    Depends: libnss3 (>= 2:3.22) but it is not installable
    Depends: libnss3 (>= 3.26) but it is not installable
    Depends: libpango-1.0-0 (>= 1.14.0) but it is not installable
    Depends: libsecret-1-0 (>= 0.18) but it is not installable
    Depends: libstdc++6 (>= 5) but it is not installable
    Depends: libstdc++6 (>= 5.2) but it is not installable
    Depends: libstdc++6 (>= 6) but it is not installable
    Depends: libx11-6 but it is not installable
    Depends: libx11-6 (>= 2:1.4.99.1) but it is not installable
    Depends: libxcb1 (>= 1.9.2) but it is not installable
    Depends: libxcomposite1 (>= 1:0.4.4-1) but it is not installable
    Depends: libxdamage1 (>= 1:1.1) but it is not installable
    Depends: libxext6 but it is not installable
    Depends: libxfixes3 but it is not installable
    Depends: libxkbcommon0 (>= 0.4.1) but it is not installable
    Depends: libxkbfile1 but it is not installable
    Depends: libxrandr2 but it is not installable
    Recommends: libvulkan1 but it is not installable
I mean, I understand nothing. I would just like to set up local web dev environment to start coding after codecademy course and web says my Chromebook should install VS Code and all other dependencies.

Pleeease if anyone could advice i would be more than grateful! karo

**Comment by shgrrts** [comment]
sudo apt-get -f install changes nothing

**Reply by shgrrts** [reply]
go_girl@penguin:~$ sudo apt-get -f install
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 41 not upgraded.

**Comment by shgrrts** [comment, marked as answer]
ok so I removed Linux dev environment and  installed it again, adding sudo apt-get -f install and then following the steps from here https://linuxize.com/post/how-to-install-visual-studio-code-on-debian-10/ on how to install VS Code on Debian10. No idea if thats what i was meant to do, but works. 

If i double click on the file from the downloads, the errors are still same as before. I launched VS Code from the command line with 'code'. 

If anyone has any idea what just happened feel free to share.

**Comment by Jobians** [comment]
Hi, I fixed my own by following this article step by step 
https://www.codecademy.com/article/web-programming-on-chromebook

Reason the error occur because you downloaded wrong vs code version for your Chromebook 

![Screenshot_2023-12-03-21-30-24-57_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/microsoft/vscode-discussions/assets/88005779/d4d208b4-5fe5-40bb-b13f-1e7fda326bc9)

**Reply by riddhijani11794** [reply]
voila! thank you, it helps! thanks a ton!

**Reply by Tikursew00** [reply]
Thanks ran into the same issue as the previous guy above. Lifesaver.

**Reply by Grecco-the-crewman** [reply]
This was SOOOO helpful. Thank you so much
